File: Data Structures & Algorithms/partition-equal-subset-sum/submission-16.java
public class Solution {
    public boolean canPartition(int[] nums) {
        int total = 0;
        for (int num : nums) {
            total += num;
        }
        if (total % 2 != 0) {
            return false;
        }

        int target = total / 2;
        int dp = 1;

        for (int num : nums) {
            dp |= dp << num;
        }

        return (dp & (1 << target)) != 0;
    }
}

File: Data Structures & Algorithms/single-number/submission-16.cs
public class Solution {
    public int SingleNumber(int[] nums) {
        Array.Sort(nums);
        int i = 0;
        while (i < nums.Length - 1) {
            if (nums[i] == nums[i + 1]) {
                i += 2;
            } else {
                return nums[i];
            }
        }
        return nums[i];
    }
}